packages/manifold/contracts/edition/ManifoldERC721Edition.sol

```diff
@@ -29,101 +29,102 @@ contract ManifoldERC721Edition is CreatorExtension, ICreatorExtensionTokenURI, I
-        require(count > 0, "Invalid amount requested");
-        require(_totalSupply[creator][series]+count <= _maxSupply[creator][series], "Too many requested");
-        
-        uint256[] memory tokenIds = IERC721CreatorCore(creator).mintExtensionBatch(recipient, count);
-        _updateIndexRanges(creator, series, tokenIds[0], count);
+    function mint(address creatorCore, uint256 instanceId, uint256 currentSupply, IManifoldERC721Edition.Recipient[] memory recipients) external override nonReentrant creatorAdminRequired(creatorCore) {        
+        if (_totalSupply[creatorCore][instanceId] != currentSupply) revert("Incorrect supply");
+        _mintTokens(creatorCore, recipients, instanceId);
+    }
+
+    function _mintTokens(address creatorCore, Recipient[] memory recipients, uint256 instanceId) private {
+        if (_totalSupply[creatorCore][instanceId]+1 > _maxSupply[creatorCore][instanceId]) revert("Too many requested");
+        if (recipients.length == 0) revert("No recipients");
+
+        // Grab the start index by minting off a first token...
+        uint256 startIndex = IERC721CreatorCore(creatorCore).mintExtension(recipients[0].recipient);
+        --recipients[0].count; // will revert with underflow if you accidentally have a count of 0... this is good (I think)
+        uint count = 1;
+        for (uint256 i; i < recipients.length;) {
+            count += recipients[i].count;
+            if (_totalSupply[creatorCore][instanceId]+count > _maxSupply[creatorCore][instanceId]) revert("Too many requested");
+            IERC721CreatorCore(creatorCore).mintExtensionBatch(recipients[i].recipient, recipients[i].count);
+            unchecked{++i;}
+        }
+
+        _updateIndexRanges(creatorCore, instanceId, startIndex, count);
     }
 
     /**
-     * @dev See {IManifoldERC721Edition-mint}.
+     * @dev See {IManifoldERC721Edition-createSeries}.
      */
-    function mint(address creator, uint256 series, address[] calldata recipients) external override nonReentrant creatorAdminRequired(creator) {
-        require(recipients.length > 0, "Invalid amount requested");
-        require(_totalSupply[creator][series]+recipients.length <= _maxSupply[creator][series], "Too many requested");
-        
-        uint256 startIndex = IERC721CreatorCore(creator).mintExtension(recipients[0]);
-        for (uint256 i = 1; i < recipients.length;) {
-            IERC721CreatorCore(creator).mintExtension(recipients[i]);
-            unchecked{i++;}
-        }
-        _updateIndexRanges(creator, series, startIndex, recipients.length);
+    function createSeries(address creatorCore, uint256 maxSupply_, string calldata prefix, uint256 instanceId, Recipient[] memory recipients) external override creatorAdminRequired(creatorCore) returns(uint256) {
+        if (instanceId == 0 || maxSupply_ == 0 || _maxSupply[creatorCore][instanceId] != 0) revert("Invalid instance");
```

Comment:
as per above, change to (also no need to return anything anymore) (also argument ordering to be consistent with other extensions, creatorCore and instanceId first, because they are the keys)
```
    function createSeries(address creatorCore, uint256 instanceId, uint24 maxSupply, string calldata prefix, Recipient[] memory recipients) external override creatorAdminRequired(creatorCore) {
    if (instanceId > MAX_UINT_56 || maxSupply == 0 || _editionInfo[creatorCore][instanceId].maxSupply != 0) revert InvalidInput();
    uint8 creatorContractVersion;
        try IERC721CreatorCoreVersion(creatorContractAddress).VERSION() returns(uint256 version) {
            require(version <= 255, "Unsupported contract version");
            creatorContractVersion = uint8(version);
        } catch {}
    _editionInfo[creatorCore][instanceId] = EditionInfo({
         maxSupply: maxSupply,
         tokenPrefix: tokenPrefix,
         contractVersion: creatorContractVersion,
    })
        
 ```